install_config/configuring_authentication.adoc

```diff
@@ -65,9 +65,64 @@ Basic authentication challenges are only sent if a `X-CSRF-Token` header is
 present on the request. Clients that expect to receive Basic `WWW-Authenticate`
 challenges should set this header to a non-empty value.
 
-.^|`login`     | When *true*, unauthenticated token requests from web clients
+|`login`     | When *true*, unauthenticated token requests from web clients
 (like the web console) are redirected to a login page backed by this provider.
 Not supported by all identity providers.
+
+|`mappingMethod`  | Defines how new identities are mapped to users when they login. See link:#mappingmethod-parameter[below] for more information.
+|===
+
+[NOTE]
+If you change authentication identity providers, but want to keep the same set
+of users, ensure the new provider has the same `*name*` setting. This maintains
+any user settings between authentication methods. Changing the `*name*` setting
+when changing authentication providers can result in multiple users with the
+same configuration.
+
+[[mappingmethod-parameter]]
+
+=== The `*mappingMethod*` Parameter
+
+Setting the `*mappingMethod*` parameter in a
+link:../install_config/master_node_configuration.html[master configuration file]
+determines how identities are mapped to users:
+
+====
+----
+...
+oauthConfig:
+  identityProviders:
+  - challenge: false
+    login: false
+    mappingMethod: "claim"
+...
+----
+====
+
+When set to the default `*claim*` value, OAuth will fail if the identity is
+mapped to a previously-existing user name. The following table outlines the use
+cases for the available `*mappingMethod*` parameter values:
+
+[cols="2,8"]
+|===
+|Parameter  | Description
+
+|`*claim*` | The default value. Provisions a preferred identity for the user.
+Fails if the preferred user name is mapped to an already-existing identity.
+
+|`*lookup*` | The value for the cluster administrator to manually configure
+identities. This will only find existing identities, and not create them.
+
+|`*generate*` | Provision unique user names generated from the user name of an
+existing identity. For example, create "user2" from "user". Do not use this
+setting when using external processes that require exact matches between
+OpenShift user names and the identity provider's user names. For example, LDAP
+group sync.
+
+|`*add*` | Provision the preferred user name for the identity, as well as add
+the user, if the user name is already mapped to another identity. Necessary when
+multiple identity providers are configured that identify the same set of users
+and map to the same user names.
```

Comment:
Provisions a user with the identity's preferred user name. If a user with that user name already exists, the identity is mapped to the existing user, adding to any existing identity mappings for the user. Needed when multiple identity providers are configured which identity the same set of users and map to the same usernames.